unit/api/base_resource.py

```diff
@@ -8,9 +8,29 @@
 MAX_DURATION = 300
 
 
-def fatal_code(e):
+def backoff_handler(e):
     code = e.status_code
-    return 500 <= code < 600 or code == 408 or code == 429
+    return (is_timeout(code) or is_rete_limit(code) or is_server_error(code)) and idempotency_key_is_not_present(e)
```

Comment:
typo - is_rate_limit

Reply:
shouldn't be and **not** `idempotency_key_is_not_present` or `idempotency_key_is_present` - we want to allow retry only when there's idempotency key, right?

Reply:
ohh, rete is in French